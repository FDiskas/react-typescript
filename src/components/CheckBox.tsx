import * as React from 'react';

import * as styles from './CheckBox.css';

export interface IItem {
    id: number;
    label: string;
    checked?: boolean;
}

export interface IProps extends IItem {
    handleClick: (item: IItem, checked: boolean) => void;
}

interface ICheckBoxState {
    checked: boolean;
}

export class CheckBox extends React.Component<IProps, ICheckBoxState> {
    constructor(props: IProps) {
        super(props);

        this.state = {
            checked: props.checked || false,
        };
    }

    handleClick = () => {
        const checked = !this.state.checked;
        this.setState({ checked });

        const { id, label } = this.props;
        this.props.handleClick({ id, label }, checked);
    };

    public render() {
        const { id, label } = this.props;

        return (
            <div key={id} onClick={this.handleClick} className={styles.checkBox}>
                {id}. {!this.state.checked ? '[ ] ' : '[x] '}{label}
            </div>
        );
    }
}
