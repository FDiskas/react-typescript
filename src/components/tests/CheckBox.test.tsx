import * as React from 'react';
import { shallow } from 'enzyme';
import enzymeToJson from 'enzyme-to-json';

import { CheckBox, IItem } from '../CheckBox';

describe('CheckBox.js', () => {
    it('renders without crashing', () => {
        const handleClick = (item: IItem, checked: boolean) => jest.fn();
        shallow(<CheckBox handleClick={handleClick} id={1} label={'Testing'} />);
    });

    it('matches snapshot', () => {
        const handleClick = (item: IItem, checked: boolean) => jest.fn();
        const wrapper = shallow(<CheckBox handleClick={handleClick} id={1} label={'Testing'} />);
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('matches checked snapshot', () => {
        const handleClick = jest.fn();
        const wrapper = shallow(<CheckBox handleClick={handleClick} id={1} label={'Testing'} />);
        wrapper.find('div').first().simulate('click');
        expect(enzymeToJson(wrapper)).toMatchSnapshot();
    });

    it('handle click', () => {
        const handleClick = jest.fn();
        const wrapper = shallow(<CheckBox handleClick={handleClick} id={1} label={'Testing'} />);
        wrapper.find('div').first().simulate('click');
        expect(handleClick).toBeCalledWith({ id: 1, label: 'Testing' }, true);
    });
});
