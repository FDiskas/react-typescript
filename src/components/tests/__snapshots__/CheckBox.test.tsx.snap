// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox.js matches checked snapshot 1`] = `
<div
  className="checkBox"
  onClick={[Function]}
>
  1
  . 
  [x] 
  Testing
</div>
`;

exports[`CheckBox.js matches snapshot 1`] = `
<div
  className="checkBox"
  onClick={[Function]}
>
  1
  . 
  [ ] 
  Testing
</div>
`;
